src/api/api.ts

```diff
@@ -0,0 +1,43 @@
+import axios, { type AxiosInstance } from 'axios'
+
+const BASE_URL = import.meta.env.VITE_APP_BASE_URL
+
+// API Response 공통 타입
+export interface ApiResponse<T> {
+  data?: T
+  error?: {
+    code: string
+    message: string
+    details?: Array<{ field: string; message: string }>
+  }
+  success: boolean
+}
+
+export const instance: AxiosInstance = axios.create({
+  baseURL: import.meta.env.VITE_APP_BASE_URL,
+  withCredentials: true,
+  headers: {
+    'Content-Type': 'application/json',
+    //  Authorization: `${getAccessTokenLocalStorage()}`
+  },
+})
+
+export function get<T>(...args: Parameters<typeof instance.get>) {
+  return instance.get<T, T>(...args)
+}
+
+export function post<T>(...args: Parameters<typeof instance.post>) {
+  return instance.post<T>(...args)
+}
+
+export function put<T>(...args: Parameters<typeof instance.put>) {
+  return instance.put<T>(...args)
+}
+
+export function patch<T>(...args: Parameters<typeof instance.patch>) {
+  return instance.patch<T, T>(...args)
+}
+
+export function del<T>(...args: Parameters<typeof instance.delete>) {
```

Comment:
의미를 명확하게 나타내는 완전한 이름을 사용하는게 더 좋아보이긴 합니다!

```suggestion
export function delete<T>(...args: Parameters<typeof instance.delete>) {
```

Reply:
delete는 예약어라서 사용할 수 없어서 del로 정했습니다ㅠ_ㅠ

Reply:
아하 그렇군요!!! 넵~~